Extension/src/LanguageServer/settings.ts

```diff
@@ -996,24 +919,51 @@ export class OtherSettings {
         this.resource = resource;
     }
 
-    public get editorTabSize(): number | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<number>("tabSize"); }
-    public get editorInsertSpaces(): boolean | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<boolean>("insertSpaces"); }
-    public get editorAutoClosingBrackets(): string | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<string>("autoClosingBrackets"); }
-    public get filesEncoding(): string | undefined { return vscode.workspace.getConfiguration("files", { uri: this.resource, languageId: "cpp" }).get<string>("encoding"); }
-    public get filesAssociations(): any { return vscode.workspace.getConfiguration("files").get("associations"); }
-    public set filesAssociations(value: any) {
-        void vscode.workspace.getConfiguration("files").update("associations", value, vscode.ConfigurationTarget.Workspace);
+    private getVSCodeSettingAsString(setting: string, configuration: string, resource: any): string | undefined {
+        const fullConfiguration = vscode.workspace.getConfiguration(setting, resource);
+        const value = fullConfiguration.get<string>(configuration);
+        if (isString(value)) {
+            return value;
+        }
+        const config = fullConfiguration.inspect<string>(configuration);
+        return config?.defaultValue;
```

Comment:
In the places where this is used, it looks like we hard code a default for when it returned undefined. I think that should move here and then this function will always return a definitive answer.

`config?.defaultValue ?? <default>`

Reply:
Sorry, this comment was meant for `editorTabSize`.